test/hotspot/jtreg/compiler/codegen/aes/Test8299817.java

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2023 SAP SE. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+/**
+ * @test
+ * @key randomness
+ * @bug 8299817
+ * @summary AES-CTR cipher failure with multiple short (< 16 bytes) update calls.
+ * @library /test/lib /
+ * @build jdk.test.whitebox.WhiteBox
+ * @run driver jdk.test.lib.helpers.ClassFileInstaller jdk.test.whitebox.WhiteBox
+ *
+ * @run main/othervm -Xbatch
+ * -XX:+UnlockDiagnosticVMOptions -XX:+WhiteBoxAPI -Xbootclasspath/a:.
+ * compiler.codegen.aes.Test8299817
+ */
+
+package compiler.codegen.aes;
+
+import java.util.Arrays;
+import java.util.Random;
+import javax.crypto.Cipher;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+import compiler.whitebox.CompilerWhiteBoxTest;
+import jdk.test.whitebox.code.Compiler;
+import jdk.test.lib.Utils;
+import jtreg.SkippedException;
+
+public class Test8299817 {
+    private static final String ALGO = "AES/CTR/NoPadding";
+    private static final int LOOPS        = 20000;
+    private static final int WARMUP_LOOPS = 10000;
+    private static final int LEN_INC   = 5;
+    private static final int LEN_STEPS = 13;
+    private static final int LEN_MAX   = LEN_INC*LEN_STEPS;
+    private static final int SEG_INC   = 3;
+    private static final int SEG_MAX   = 11;
+    private static final int SHOW_ARRAY_LIMIT = 72;
+    private static final boolean DEBUG_MODE = false;
+
+    public static void main(String[] args) throws Exception {
+        if (!DEBUG_MODE) {
+            if (!Compiler.isIntrinsicAvailable(CompilerWhiteBoxTest.COMP_LEVEL_FULL_OPTIMIZATION, "com.sun.crypto.provider.CounterMode", "implCrypt", byte[].class, int.class, int.class, byte[].class, int.class)) {
```

Comment:
Maybe break the large lines?

Reply:
Lines split.
Thank you very much for your reviews, @TheRealMDoerr @offamitkumar @MBaesken